### ISSUE TITLE
Failing to build

### ISSUE THREAD

**Issue by gnaomo** [jam1garner/smush_info#5]
aside from replacing all `smush_discord_info` references to `smush_info_discord`
rust throws this error

```
error: expected type, found `"=r"`=====> ] 69/70: smush_info               
   --> src/lib.rs:163:46
    |
163 |     unsafe { std::arch::asm!("mov $0, x29" : "=r"(r) ::: "volatile"); }
    |                                            - ^^^^ expected type
    |                                            |
    |                                            tried to parse a type due to this type ascription                                                        
    |
    = note: `#![feature(type_ascription)]` lets you annotate an expression with a type: `<expr>: <type>`
    = note: see issue #23416 <https://github.com/rust-lang/rust/issues/23416> for more information

```
the problem may be that the syntax for asm! changed and is now different from what it used to be
(old syntax https://doc.rust-lang.org/1.12.0/book/inline-assembly.html)
(new syntax https://doc.rust-lang.org/nightly/reference/inline-assembly.html)

**Comment by gnaomo** [comment]
May be fixed with latest pull request #6 
It does not throw any error, it builds successfully and it works offline, but needs to be tested on Online Arenas.

**Comment by gnaomo** [comment]
May be fixed with latest pull request #6 
It does not throw any error, it builds successfully and it works offline, but needs to be tested on Online Arenas.
